File: src/json/control_comma_period.json.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# You can generate json by executing the following command on Terminal.
#
# $ ruby ./control_comma_dot.json.rb
#

require 'json'
require_relative '../lib/karabiner.rb'

def main
  puts JSON.pretty_generate(
    'title' => 'Change control+comma & control+period',
    'rules' => [
      {
        'description' => 'Change control+comma to comma, control+period to period',
        'manipulators' => [
          {
            'type' => 'basic',
            'from' => {
              'key_code' => 'comma',
              'modifiers' => Karabiner.from_modifiers(['control'], %w[caps_lock shift]),
            },
            'to' => [
              {
                'key_code' => 'comma',
              },
            ],
          },
          {
            'type' => 'basic',
            'from' => {
              'key_code' => 'period',
              'modifiers' => Karabiner.from_modifiers(['control'], %w[caps_lock shift]),
            },
            'to' => [
              {
                'key_code' => 'period',
              },
            ],
          },
        ],
      },
    ]
  )
end

main

File: src/json/control_comma_period.json.js
// JavaScript should be written in ECMAScript 5.1.

function main() {
  console.log(
    JSON.stringify(
      {
        title: 'Change control+comma & control+period',
        rules: [
          {
            description: 'Change control+comma to comma, control+period to period',
            manipulators: [
              {
                type: 'basic',
                from: {
                  key_code: 'comma',
                  modifiers: {
                    mandatory: ['control'],
                    optional: ['caps_lock', 'shift'],
                  },
                },
                to: [
                  {
                    key_code: 'comma',
                  },
                ],
              },
              {
                type: 'basic',
                from: {
                  key_code: 'period',
                  modifiers: {
                    mandatory: ['control'],
                    optional: ['caps_lock', 'shift'],
                  },
                },
                to: [
                  {
                    key_code: 'period',
                  },
                ],
              },
            ],
          },
        ],
      },
      null,
      '  '
    )
  )
}

main()
